content/en/blog/_posts/2022-03-31-Kubernetes-1-24-deprecations-and-removals

```diff
@@ -0,0 +1,90 @@
+* Beta or pre-release API versions must be supported for 3 releases after deprecation. 
+* Alpha or experimental API versions may be removed in any release without prior deprecation notice. 
+
+Kubernetes will continue to remove APIs by following the deprecation policy regardless of whether an API is removed due to a beta feature graduating to stable or because that API was not proven to be successful and will also make sure migration options are documented. **Deprecated** APIs have been marked for removal in a future Kubernetes release. **Removed** APIs are those that are not longer available for use in current, supported Kubernetes versions after having been deprecated. These removals have been superseded by newer, stable/generally available (GA) APIs. 
+
+## API removals, deprecations, and changes for Kubernetes 1.24
+
+* [Dynamic kubelet configuration](https://github.com/kubernetes/enhancements/issues/281): `DynamicKubeletConfig` is used to enable the dynamic configuration of kubelet. The `DynamicKubeletConfig` flag was deprecated in Kubernetes 1.22. As of 1.24, the feature gate was removed from kubelet. See [Reconfigure kubelet](/docs/tasks/administer-cluster/reconfigure-kubelet/). Refer to the ["Dynamic kubelet config is removed" KEP](https://github.com/kubernetes/enhancements/issues/281) for more information. 
+* [Dynamic log sanitization](https://github.com/kubernetes/kubernetes/pull/107207): The experimental dynamic log sanitization feature is deprecated as of 1.24. This feature introduced a logging filter that could be applied to all Kubernetes system components logs to prevent various types of sensitive information from leaking via logs. An alternative approach to ensuring secrets are not being logged, one which implemented static analysis as part of a presubmit CI graduated to stable in 1.23. Refer to [KEP-1753: Kubernetes system components logs sanitization](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1753-logs-sanitization#deprecation) for more information. 
+* In-tree provisioner to CSI driver migration: This applies to a number of in-tree plugins, including [Portworx](https://github.com/kubernetes/enhancements/issues/2589). Refer to the [In-tree Storage Plugin to CSI Migration Design Doc](https://github.com/kubernetes/design-proposals-archive/blob/main/storage/csi-migration.md#background-and-motivations) for more information. 
+* [Removing Dockershim from kubelet](https://github.com/kubernetes/enhancements/issues/2221) (**more on this below**): the Container Runtime Interface (CRI) for Docker (i.e. Dockershim) is currently a built-in container runtime in the kubelet code base. It was deprecated in 1.20. As of 1.24, dockershim has been removed from kubelet. Check out this blog on [what you need to do to have a working cluster for 1.24](https://github.com/kubernetes/website/issues/30917). 
```

Comment:
@mickeyboxell this is *STILL* not an API removal, deprecation or change.

The heading for this section is correct; you should move the changes that are not API-related and put them under a different heading.

Reply:
You're right. I'll remove the word API.

Reply:
maybe i'm missing something, but this link `[what you need to do to have a working cluster for 1.24](https://github.com/kubernetes/website/issues/30917).` does not point to a blog, but an issue that a blog should be created, maybe this needs to be corrected when the blog is published?